superduperdb/components/model.py

```diff
@@ -190,9 +193,17 @@ def predict(
         if isinstance(select, dict):
             select = Serializable.deserialize(select)
 
-        if db is not None:
-            db.metadata.add_query(select, self.identifier)
+        if isinstance(select, Table):
+            select = select.to_query()
 
+        if db is not None:
+            if isinstance(select, IbisCompoundSelect):
+                try:
+                    _ = db.metadata.get_query(str(hash(json.dumps(select.serialize()))))
```

Comment:
lets keep it like
db.metadata.get_query(str(hash(json.dumps(select.serialize())))) without `_`

also 

how about select.hash() which abstracts this messy line

Reply:
Yes agreed.